File: Tests/executors/CSharp/CoreExecutors/UploadChunkedExecutor.cs
using Framework;
using Framework.Utility;

public class UploadChunkedExecutor : HeavyOperationStatelessExecutor
{
	public override void DoPost(HttpRequest request, HttpResponse response)
	{
		using FileStream stream = new(request.GetHeaders()["File-Name"], FileMode.Create);
		using BinaryWriter writer = new(stream);
		List<byte[]> chunks = [.. request.GetChunks()];

		foreach(byte[] chunk in chunks)
		{
			writer.Write(chunk);
		}

		response.SetResponseCode(ResponseCodes.Created);
		response.SetBody("Finish uploading file");
	}
}

File: Tests/executors/CXX/core/src/UploadChunkedExecutor.cpp
#include "UploadChunkedExecutor.h"

#include <fstream>

void UploadChunkedExecutor::doPost(framework::HttpRequest& request, framework::HttpResponse& response)
{
	std::ofstream stream(request.getHeaders().at("File-Name"), std::ios::binary);

	for (const std::string& chunk : request.getChunks())
	{
		stream.write(chunk.data(), chunk.size());
	}

	response.setResponseCode(framework::ResponseCodes::created);
	response.setBody("Finish uploading file");
}

DEFINE_EXECUTOR(UploadChunkedExecutor)
